server/database/repositories/TournamentRepository.ts

```diff
@@ -71,7 +72,7 @@ export default class TournamentRepository {
   }
 
   public static async updateMatch(match: MatchWithoutMS) {
-    if (!match.id) throw new AppError('Provided match does not contain id', 400);
+    if (!match.id || !isValidObjectId(match.id)) throw new AppError('Provided match does not contain id', 400);
```

Comment:
I thought we agreed that, when we want to update match, we have intention to update existing match, otherwise it makes no sens. Also what does null means when returned from that function? It could mean error during update, or invalid ID. I find it simple to return nothing, It is different with getFunctions, because we want info when nothing with certain ID is there. This is just my opinion, if this is generating problems, feel free and change it. 
In the above listing if statement is checking if match has id property ( because i can not ) so we say it is error, and that tells programmer, that this function suppose to work with match with id (fetched from db) .

Reply:
okej but, if you want to update no exiting resource you shoud get 404 http status. 
404:
> HTTP 404 (Not Found) is usually returned when a target resource does not exist or when the server doesn't want to disclose that one exists.


400:
>The request could not be understood by the server due to malformed syntax. The client SHOULD NOT repeat the request without modifications.

Internet says we should return 404, i want to handle null to return 404 oraz return 404 here in this line

Reply:
ok